### DISCUSSION TITLE
Possibility to specify redirect behaviour in `Auth` instance?

### DISCUSSION THREAD

**Discussion by septatrix** [encode/httpx#1914] [General] [answered]
Currently this does not seem to be possible as [the method(s) handling authentication](https://github.com/encode/httpx/blob/62b1666dc6c6b012b258574c4239b8ff8f481619/httpx/_client.py#L890) use the clients `follow_redirects` parameter and the auth flow is only able to "communicate" using `Request` objects which do not carry any redirect hints. In some cases this might however be necessary.

**Comment by lovelydinosaur** [comment]
You'd need to explain why you want to do that? It's not really clear from the question.

**Reply by septatrix** [reply]
I have a complex authentication flow which requires sending multiple requests. Some of these will have to follow a redirect and some of them should not. This is due to the fact that the redirected requests also have to be modified and the final redirect should not be send/followed. While the former is still possible manually (by checking `has_redirect` und yielding `next_request`) the latter seems impossible to do. I guess I could use some more involved logic by also using hooks but I would like everything to be inside an auth handler.

**Comment by septatrix** [comment, marked as answer]
I know fixed this problem by manually calling an authentication handler first and afterwards using the received cookie+token in the following requests: https://github.com/septatrix/pymoodle/blob/a19cdf6a9b1247ed63c34d8dd7e73bada5ac044c/moodle/contrib/identityproviders.py#L72
